langlib/lang.__internal/src/main/java/org/ballerinalang/langlib/internal/GetElementType.java

```diff
@@ -42,16 +43,23 @@ public static BTypedesc getElementType(Object td) {
     }
 
     private static BTypedesc getTypeDescValue(Type type) {
-        if (type.getTag() == TypeTags.ARRAY_TAG) {
-            return ValueCreator.createTypedescValue(((ArrayType) type).getElementType());
-        } else if (type.getTag() == TypeTags.STREAM_TAG) {
-            return ValueCreator.createTypedescValue(((StreamType) type).getConstrainedType());
-        } else if (type.getTag() == TypeTags.TABLE_TAG) {
-            return ValueCreator.createTypedescValue(((TableType) type).getConstrainedType());
-        } else if (type.getTag() == TypeTags.FINITE_TYPE_TAG && ((FiniteType) type).getValueSpace().size() == 1) {
-            return getTypeDescValue(((BValue) (((FiniteType) type).getValueSpace().iterator().next())).getType());
+        switch (type.getTag()) {
+            case TypeTags.ARRAY_TAG:
+                return ValueCreator.createTypedescValue(((ArrayType) type).getElementType());
+            case TypeTags.TUPLE_TAG:
+                return ValueCreator.createTypedescValue(
+                        TypeCreator.createUnionType(((TupleType) type).getTupleTypes()));
+            case TypeTags.STREAM_TAG:
+                return ValueCreator.createTypedescValue(((StreamType) type).getConstrainedType());
+            case TypeTags.FINITE_TYPE_TAG:
+                if (((FiniteType) type).getValueSpace().size() == 1) {
+                    return getTypeDescValue(
+                            ((BValue) (((FiniteType) type).getValueSpace().iterator().next())).getType());
+                }
+                break;
+            default:
+                break;
```

Comment:
Should this be an error?

Reply:
Due to `public isolated function getElementType(typedesc<Type[]> | typedesc<stream<Type, CompletionType>> td)` I think only the above 4 types are allowed to come inside the function. So `return ValueCreator.createTypedescValue(PredefinedTypes.TYPE_NULL);` is never executed anyway. Should I throw an error instead of that line like https://github.com/ballerina-platform/ballerina-lang/pull/32257/files#r697291327 or may I remove the line?

Reply:
@pcnfernando Can you please tell why the `(((FiniteType) type).getValueSpace().size() == 1)` is required. In the samples I tested, the cases which came under the `FINITE_TYPE_TAG` had this property. Are there any case which does not have this property?

What I am trying to do is to increase the code coverage, change the `case TypeTags.FINITE_TYPE_TAG:` to the `default case` and only put `return getElementTypeDescValue(((BValue) (((FiniteType) type).getValueSpace().iterator().next())).getType());` there. I would like to put `assert ((((FiniteType) type).getValueSpace().size() == 1) && type.getTag() == TypeTags.FINITE_TYPE_TAG);` but that does not get executed without the `-enableassertions` java cli option.

Reply:
@pcnfernando Could you please tell why the `if (((FiniteType) type).getValueSpace().size() == 1)` condition is required. In the samples I tested which came under the case `FINITE_TYPE_TAG`, all of them had this property. Is there any case which does not have that property?

What I am trying to do here is to increase the code coverage changing the `case TypeTags.FINITE_TYPE_TAG:` to `default case` and put `return getElementTypeDescValue(
                            ((BValue) (((FiniteType) type).getValueSpace().iterator().next())).getType());` under that. I would like to put `assert ((((FiniteType) type).getValueSpace().size() == 1) && type.getTag() == TypeTags.FINITE_TYPE_TAG);` before that but that does not get executed without `-enableassertions` java clip option.

Reply:
Finite case itself was added to handle the change introduced for typedesc with immutable values in https://github.com/ballerina-platform/ballerina-lang/pull/32105. Hence, we can assume its value space is always 1. Please add a comment for finite case saying that it's reached only for immutable values.
Since this method should accept array/tuple and streams alone through ballerina, no need to throw the error. Use stream type as the default case.

Reply:
I will make these changes. Thank you!